controllers/mdutil/util.go

```diff
@@ -521,7 +536,7 @@ func NewMSNewReplicas(deployment *clusterv1.MachineDeployment, allMSs []*cluster
 			return 0, err
 		}
 		// Find the total number of machines
-		currentMachineCount := GetReplicaCountForMachineSets(allMSs)
+		currentMachineCount := GetTotalReplicaCountForMachineSets(allMSs)
```

Comment:
If we can assume that a MachineSet's status.replicas includes deleting or not yet ready Machines, then it seems like it should be okay to use GetActualReplicaCountForMachineSets rather than needing to take the max here.

Reply:
The max is for the case that, for two machinesets, one spec=0,status=2, the other spec=1,status=0. Then at this moment total count should be 3 instead of 2.

Reply:
Gotcha, I'm wondering if we need to avoid increasing the spec for the new machineSet until things have finished ensuring that the old machineSet has finished stabilizing if increasing would exceed the max surge...  I guess I'm wondering if this is the correct place to try to do that enforcement, especially since using the max of spec.replicas and status.replicas seems to be more of an approximation of trying to do the right thing vs giving assurance that we are doing the right thing.

Reply:
I'm trying to understand this. Do you mean we could allow exceed max surge here in MS.spec, and do enforce within MS controller that won't actually create the Machine object if it already exceed surge? If this understanding is correct, I'm not sure how MS controller will be able to do that, because maxSurge and total number of replicas only available in MD.

Reply:
I think Jason was suggesting if we need a check in the webhook validator to ensure that the status is up to date before allowing any type of resize. Probably worth a different issue though